lenskit-core/src/main/java/org/lenskit/rerank/CandidateItemSelector.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * LensKit, an open source recommender systems toolkit.
+ * Copyright 2010-2016 LensKit Contributors.  See CONTRIBUTORS.md.
+ * Work on LensKit has been funded by the National Science Foundation under
+ * grants IIS 05-34939, 08-08692, 08-12148, and 10-17697.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package org.lenskit.rerank;
+
+import org.lenskit.api.Result;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+/**
+ * Interface for classes that select the next item to recommend from a list of candidate items. This class
+ * is used in {@link GreedyRerankingItemRecommender} to define the greedy strategy for selecting the next item.
+ * Often this will be a search for the best item that satisfies some constraint or the best item by some metric that
+ * takes in account what items have already been recommended.
+ *
+ * @author Daniel Kluver
+ */
+
+
+public interface CandidateItemSelector {
```

Comment:
I'm not in love with this name - it overloads with `ItemSelector` (not a big problem), and with the yet-to-be-introduced `CandidateSelector` that will abstract the notion of identifying a default set of candidates for a recommend operation.

What if we call it `GreedyRerankStrategy`, so that the name indicates clearly the use of the Strategy pattern?